src/app-layer-smtp.c

```diff
@@ -985,7 +985,10 @@ static int SMTPParseCommandBDAT(SMTPState *state, const SMTPLine *line)
     }
     memcpy(strbuf, line->buf + i, len);
     strbuf[len] = '\0';
-    if (StringParseUint32(&state->bdat_chunk_len, 10, 0, strbuf) < 0) {
+    char **endptr = NULL;
+    errno = 0;
+    state->bdat_chunk_len = strtoul((const char *)strbuf, (char **)&endptr, 10);
+    if (((uint8_t *)endptr == line->buf + i) || errno == ERANGE) {
```

Comment:
len instead of i ?

Reply:
Actually, i think it should be `strbuf == *endptr`

Reply:
Maybe Suricata should add a wrapper around `strtoul` with a comprehensive way to return error...